src/utils.rs

```diff
@@ -11,19 +11,20 @@
 * SPDX-License-Identifier: Apache-2.0
 ********************************************************************************/
 
-use anyhow::{Context, Result};
+use anyhow::{anyhow, Context, Ok, Result};
 use console::Term;
-use hdrhistogram::Histogram;
 use log::debug;
 use serde_json::from_reader;
 use std::{cmp::max, fs::OpenOptions, io::Write, time::SystemTime};
 
 use crate::{
-    config::{Config, Signal},
-    measure::Api,
+    config::{Config, Group, Signal},
+    measure::MeasurementResult,
 };
 
-pub fn read_config(config_file: Option<&String>) -> Result<Vec<Signal>> {
+const MAX_NUMBER_OF_GROUPS: usize = 10;
```

Comment:
Why do we want a limit here?

Reply:
It was a acceptance criteria to limit it.